pkg/runtime/reconciler.go

```diff
@@ -830,11 +834,32 @@ func (r *resourceReconciler) handleRequeues(
 			}
 			// The code below only executes for "ConditionTypeResourceSynced"
 			if condition.Status == corev1.ConditionTrue {
-				if duration := r.rmf.RequeueOnSuccessSeconds(); duration > 0 {
+				duration := r.rmf.RequeueOnSuccessSeconds()
+				if duration > 0 {
```

Comment:
I think I prefer the original condition. If `duration < 0`, we should just assume it `= 0`.